azavea/temperate
# Upgrade Angular, Node.js, and related dependencies

## Overview

The main effort in this PR is to upgrade Angular from version `4` to `8`.

As a consequence of that effort, the following other changes were also made:
 - Node.js upgraded from `6` to `10`
 - `ngx-bootstrap` updated to the latest version, instead of using our forked version
 - Bootstrap updated from version `3` to `4`
 - The `climate-change-components` repository was copied into this repository, instead of remaining a separate dependency, due to difficulties in upgrading and integrating with the new Angular CLI support for library projects
 - The `difflib` dependency (used by the password validation popup) was copied into this project with minor fixes, due to an error which was not fatal in our old CLI toolchain but became so in the newer toolchain.


### Demo

N/A

Everything should continue to work as it did before, with no changes in behavior or styling.


## Testing Instructions

 * `scripts/update`
 * `scripts/server`
 * Test _everything_ in the front-end application (you should be able to skip the Django Admin site). Make a small change and verify that live reload works.
 * Access the production build at `localhost:8000` by stopping `scripts/server` and running `docker-compose up django nginx`
 * Test everything in the application again using the production build.


Closes #1202 
Closes #263 
Closes #876
